lib/dal/src/attribute/prototype/argument.rs

```diff
@@ -478,29 +475,39 @@ impl AttributePrototypeArgument {
         Ok(None)
     }
 
+    /// This sets the value source on an APA.
+    /// If the APA already has a value source, it will be replaced.
     pub async fn set_value_source(
         ctx: &DalContext,
         apa_id: AttributePrototypeArgumentId,
         value_source: impl Into<ValueSource>,
     ) -> AttributePrototypeArgumentResult<()> {
+        // First remove any existing sources.
         let workspace_snapshot = ctx.workspace_snapshot()?;
-
-        for existing_value_source in workspace_snapshot
-            .outgoing_targets_for_edge_weight_kind(
+        workspace_snapshot
+            .remove_outgoing_edges_of_kind(
                 apa_id,
                 EdgeWeightKindDiscriminants::PrototypeArgumentValue,
             )
-            .await?
-        {
-            workspace_snapshot
-                .remove_edge(
-                    apa_id,
-                    existing_value_source,
-                    EdgeWeightKindDiscriminants::PrototypeArgumentValue,
-                )
-                .await?;
-        }
+            .await?;
+        workspace_snapshot
```

Comment:
There was an issue here where set_value_source() would not remove the old subscription.